curated_transformers/models/refined_web_model/causal_lm.py

```diff
@@ -0,0 +1,75 @@
+from typing import Any, List, Mapping, Optional, Type, TypeVar
+
+import torch
+from torch import Tensor
+from torch.nn import Linear
+
+from ..attention import AttentionMask
+from ..hf_hub import FromPretrainedHFModel
+from ..module import CausalLMModule
+from ..output import CausalLMOutputWithCache, KeyValueCache
+from ._hf import convert_hf_config, convert_hf_state_dict
+from .config import RefinedWebModelConfig
+from .decoder import RefinedWebModelDecoder
+
+# Only provided as typing.Self in Python 3.11+.
+Self = TypeVar("Self", bound="RefinedWebModelCausalLM")
+
+
+class RefinedWebModelCausalLM(CausalLMModule[KeyValueCache], FromPretrainedHFModel):
+    """Refined Web Model (eg. Falcon) causal language model."""
+
+    def __init__(
+        self, config: RefinedWebModelConfig, *, device: Optional[torch.device] = None
+    ) -> None:
+        """
+        :param config: Model configuration.
+        :param device: Device on which the module is to be initialized.
+        """
+        super().__init__()
+
+        self.decoder = RefinedWebModelDecoder(config, device=device)
+        self.output_embeddings = Linear(
+            in_features=config.layer.hidden_width,
+            out_features=config.embedding.vocab_size,
+            bias=False,
```

Comment:
No `bias` term for the LM head (because of the layernorm that's performed on the last layer's output)?

Reply:
It's common to not have bias on the output embeddings on the output. I don't really like the LM head terminology, it's really more like: the model produces an embedding and you then do a matrix multiplication between the predicted embedding and the output embedding matrix to find the token for which the output embedding is the closest to the predicted embedding.